### ISSUE TITLE
Allow ReactNode to passed to values or InterpolationMap

### ISSUE THREAD

**Issue by bfricka** [i18next/react-i18next#948]
**Is your feature request related to a problem? Please describe.**
Currently, there's no way to pass `ReactNode` to `values` or in the `InterpolationMap` for `t`. This seems like a really simple implementation for `Trans`, at least, looking at the code.

**Describe the solution you'd like**
```tsx
const [t] = useTranslation();
return <div>{t('foo', {link: <a href="//tosomewhere.com">{t('to somewhere')}</a>})}</div>;
```
or
```tsx
return (
  <Trans
    i18nKey="foo"
    values={{value: `<a href="//tosomewhere.com">to somewhere</a>`}}
  />
);
```

**Describe alternatives you've considered**
There really aren't any alternatives that can work for me here. The currently supported method of using `components` with `Trans` and index-based replacement is not feasible for how our company does translations. Additionally, it's overly complicated when, for React, we can just use `React.isElement` and avoid creating those children in `mapAST` in `Trans`. This would be tougher for `useTranslation` because `t` is just `i18next.getFixedT` and thus isn't prepared to handle React-specific stuff.

I think allowing `ReactNode` for `Trans` would be a simple and very useful alternative to the tightly proscribed way that is currently supported.

**Comment by jamuhl** [comment]
possible duplicate of https://github.com/i18next/react-i18next/issues/833

**Comment by jamuhl** [comment]
PR welcome if not a duplicate of above issue / PR

**Comment by bfricka** [comment]
I would like to take a stab at this, for sure. Additionally, I think there is some complexity within `renderNodes` that can potentially be streamlined without breaking any of the logic or incurring performance penalties.  As far as using `components` vs `values`, I'll have to see. For example, having translations like: `"Sometime in the year <year>B.C.E</year>"` is strange to me compared to `"Sometime in the year {{ year }} B.C.E"`.

**Comment by jamuhl** [comment]
Guess both approaches have pros and cons:

eg. `Your <link>inbox</link> has 2 unread messages` is more readable in translation than `Your {{link}} has 2 unread messages` -> but might be a matter of taste...

**Comment by bfricka** [comment]
That's true. Plus in the `{{link}}` case, you have to translate the link text separately:
```tsx
<Trans i18nkey="yourInboxHas" values={{link: <a>t('inbox')</a>}} count={2}/>
// vs
<Trans i18nKey="yourInboxHas" components={{link: <a/>}} count={2}/>
```
I'll get a better sense of the semantics once I start working on this, but even though the first one is more verbose, it's a bit clearer to me what's going on. In the second case, the translated link text is implicitly added as a child of `<a/>`.

**Comment by jamuhl** [comment]
Another aspect...I personally would use Trans in this case like:

```js
const count = 2;

// ...

<Trans i18nKey="yourInboxHas"  count={count}>
  Your <a href="...">inbox</a> has {{ count }} unread messages
</Trans>
```
